packages/appium/lib/extension/driver-config.js

```diff
@@ -0,0 +1,215 @@
+// @ts-check
+
+import _ from 'lodash';
+import { DRIVER_TYPE } from '../constants';
+import log from '../logger';
+import { ExtensionConfig } from './extension-config';
+
+/**
+ * @extends {ExtensionConfig<DriverType>}
+ */
+export class DriverConfig extends ExtensionConfig {
+  /**
+   * A mapping of `APPIUM_HOME` values to {@link DriverConfig} instances.
```

Comment:
so this assumes we could have multiple appium homes with different configs on the same machine and switch between them?

Reply:
our tests could do this sort of thing, if they don't already.  but we needed a "key" to ensure we're not creating multiple copies of this class, and `APPIUM_HOME` was obvious, but it doesn't need to be that.  It could be an instance of a `Manifest` object, and we could use a `WeakMap`.